ForumProject/communications/views.py

```diff
@@ -1,3 +1,26 @@
-from django.shortcuts import render
+from django.contrib.auth import get_user_model
+from django.shortcuts import render, get_object_or_404
 
-# Create your views here.
+from .models import Room
+
+User = get_user_model()
+
+
+def index_view(request):
+    users = User.objects.exclude(id=request.user.id)
+    rooms_list = Room.objects.filter(name__contains=str(request.user.id))
+    return render(request, 'index.html', {
+        'rooms': rooms_list, 'users': users
+    })
+
+
+def room_view(request, user_id):
+    user = get_object_or_404(User, id=user_id)
```

Comment:
The `get_object_or_404` usage is appropriate for fetching the user based on the `user_id`. However, consider situations where the `user_id` is invalid or malformed to avoid server errors

Reply:
if the user tries to write to a person who does not exist, he receives a 404 error. 
Should it be the other way around?

Reply:
I mean the case, when`user_id` isn't a valid identifier. Trying to fetch a user with such an ID using `get_object_or_404` will raise a `ValueError` or `TypeError` before it makes any database query. You may leave this line without changes. Or you may add something like this
```
    try:
        user_id = int(user_id)  
    except ValueError:
        return HttpResponse('Invalid user ID format.', status=400)
```
before it